base/abstractarray.jl

```diff
@@ -113,10 +113,12 @@ linearindexing{T<:AbstractArray}(::Type{T}) = LinearSlow()
 linearindexing{T<:Array}(::Type{T}) = LinearFast()
 linearindexing{T<:Range}(::Type{T}) = LinearFast()
 
-*(::LinearFast, ::LinearFast) = LinearFast()
-*(::LinearSlow, ::LinearFast) = LinearSlow()
-*(::LinearFast, ::LinearSlow) = LinearSlow()
-*(::LinearSlow, ::LinearSlow) = LinearSlow()
+linearindexing(A::AbstractArray, B::AbstractArray) = linearindexing(linearindexing(A), linearindexing(B))
+linearindexing(A::AbstractArray, B::AbstractArray...) = linearindexing(linearindexing(A), linearindexing(B...))
+linearindexing(::LinearFast, ::LinearFast) = LinearFast()
```

Comment:
That would make all operations slow, if any are slow... The way it is now, some can stay fast

Sent from my iPhone

> On Jun 5, 2015, at 5:03 AM, Matt Bauman <notifications@github.com> wrote:
> 
> In base/abstractarray.jl:
> 
> > @@ -113,10 +113,12 @@ linearindexing{T<:AbstractArray}(::Type{T}) = LinearSlow()
> >  linearindexing{T<:Array}(::Type{T}) = LinearFast()
> >  linearindexing{T<:Range}(::Type{T}) = LinearFast()
> >  
> > -*(::LinearFast, ::LinearFast) = LinearFast()
> > -*(::LinearSlow, ::LinearFast) = LinearSlow()
> > -*(::LinearFast, ::LinearSlow) = LinearSlow()
> > -*(::LinearSlow, ::LinearSlow) = LinearSlow()
> > +linearindexing(A::AbstractArray, B::AbstractArray) = linearindexing(linearindexing(A), linearindexing(B))
> > +linearindexing(A::AbstractArray, B::AbstractArray...) = linearindexing(linearindexing(A), linearindexing(B...))
> > +linearindexing(::LinearFast, ::LinearFast) = LinearFast()
> I like linearindexing for this; it makes sense.  I haven't tried this, but could it be simplified with just two methods?
> 
> linearindexing(::Union(LinearFast, LinearSlow)...) = LinearSlow()
> linearindexing(::LinearFast...) = LinearFast()
> —
> Reply to this email directly or view it on GitHub.
>